lib/spack/spack/binary_distribution.py

```diff
@@ -1540,7 +1540,9 @@ def push_keys(*mirrors, **kwargs):
                 filename = fingerprint + '.pub'
 
                 export_target = os.path.join(prefix, filename)
-                spack.util.gpg.export_keys(export_target, fingerprint)
+
+                # Export public keys (private is set to False)
+                spack.util.gpg.export_keys(export_target, [fingerprint])
```

Comment:
@cosmicexplorer this is the only part that looks a little weird - putting the fingerprint into a list to run the function, which expects a list of keys. But this way, the bool for secret can be False. We could add a check for the variable type of keys and convert to list if it's a string, but since it's only used in a handful of places that's probably overkill.